frontend/javascripts/oxalis/controller/combinations/bounding_box_handlers.js

```diff
@@ -0,0 +1,279 @@
+// @flow
+import { calculateGlobalPos } from "oxalis/model/accessors/view_mode_accessor";
+import _ from "lodash";
+import { type OrthoView, type Point2, type Vector3 } from "oxalis/constants";
+import Store from "oxalis/store";
+import { getSomeTracing } from "oxalis/model/accessors/tracing_accessor";
+import Dimension from "oxalis/model/dimensions";
+import { setUserBoundingBoxBoundsAction } from "oxalis/model/actions/annotation_actions";
+import { getBaseVoxelFactors } from "oxalis/model/scaleinfo";
+import getSceneController from "oxalis/controller/scene_controller_provider";
+
+const BOUNDING_BOX_HOVERING_THROTTLE_TIME = 100;
+
+const getNeighbourEdgeIndexByEdgeIndex = {
+  // TODO: Use this to detect corners properly.
+  // The edges are indexed within the plane like this:
+  // See the distanceArray calculation as a reference.
+  //  +---0---+
+  //  |       |
+  //  2       3
+  //  |       |
+  //  +---1---+
+  //
+  "0": [2, 3],
+  "1": [2, 3],
+  "2": [0, 1],
+  "3": [0, 1],
+};
+const MAX_DISTANCE_TO_SELECTION = 15;
+
+function getDistanceToBoundingBoxEdge(
+  pos: Vector3,
+  min: Vector3,
+  max: Vector3,
+  compareToMin: boolean,
+  edgeDim: number,
+  otherDim: number,
+  planeRatio: Vector3,
+) {
+  // There are three cases how the distance to an edge needs to be calculated.
+  // Here are all cases visualized via numbers that are referenced below:
+  // Note that this is the perspective of the rendered bounding box cross section in a viewport.
+  // ---> x
+  // |  1                  1
+  // ↓    '.             .'
+  // y      ↘          ↙
+  //          +-------+
+  //          |       |
+  //    3 --> |       | <-- 3
+  //          |       |
+  //          +-------+
+  //        ↗           ↖
+  //      .'             '.
+  //     2                 2
+  //
+  // This example is for the xy viewport for y as the main direction / edgeDim.
+
+  // As the planeRatio is multiplied to the global coordinates passed to this method,
+  // the distance between the mouse and the bounding box is distorted by the factor of planeRatio.
+  // That's why we later divide exactly by this factor to let the hit box / distance
+  // between the mouse and bounding box be the same in each dimension.
+  const cornerToCompareWith = compareToMin ? min : max;
+  if (pos[edgeDim] < min[edgeDim]) {
+    // Case 1: Distance to the min corner is needed in edgeDim.
```

Comment:
> Ahhhh, now I got it :) So, in my example, edgeDim would need to be 0 because it's referring to the case where the horizontal edge is checked. And consequentially, it would compare the x values (and not the y values which I somehow assumed would also be a valid case).

Exactly :cat2: 

> Could you please adapt the above comment to explain how to interpret edgeDim? I.e., 1 means it's a vertical edge.

Not exactly, `edgeDim` gives the dimension index that the edge extents to: 0 <=> edge direction is x (like above), 1 <=> edge direction is y (like above but downwards), 2 <=> edge direction is z.

I changed the comment highlighted above in the code. Could you please check whether this is now understandable? If not, could you please try to help me explain this logic? I think the explanation is not straightforward.